eldritch/test_events.py

```diff
@@ -1167,6 +1167,199 @@ def testCastAndGoInsane(self):
+    self.assertFalse(buy.is_resolved())
+    self.assertFalse(self.char.possessions)
+    food = items.Food()
+    self.state.common.append(food)
+
+    self.state.event_stack.append(buy)
+    choice = self.resolve_to_choice(CardChoice)
+    self.assertEqual(choice.choices, ["Food", "Nothing"])
+    choice.resolve(self.state, "Nothing")
+    self.resolve_until_done()
+
+    self.assertTrue(buy.is_resolved())
+    self.assertEqual(self.char.dollars, 3)
+    self.assertFalse(self.char.possessions, [])
+    self.assertEqual(len(self.state.common), 1)
+    self.assertEqual(self.state.common[0], food)
+
+  def testPurchaseOneAtListPoor(self):
+    draw = Purchase(self.char, "common", 1)
+    self.char.dollars = 0
+    self.assertFalse(draw.is_resolved())
+    self.assertFalse(self.char.possessions)
+    food = items.Food()
+    self.state.common.append(food)
+
+    # When you are too poor to buy the item, you do not get a choice.
+    self.state.event_stack.append(draw)
+    self.resolve_until_done()
+
+    self.assertTrue(draw.is_resolved())
+    self.assertEqual(len(self.state.common), 1)
+    self.assertEqual(self.state.common[0], food)
+    self.assertFalse(self.char.possessions)
+
+  def testPurchaseTwoAtListAffordOne(self):
+    buy = Purchase(self.char, "common", 2, keep_count=2)
+    self.char.dollars = 3
+    self.assertFalse(buy.is_resolved())
+    self.assertFalse(self.char.possessions)
+    food = items.Food()
+    gun = items.TommyGun()
+    self.state.common.extend([gun, food])
+
+    self.state.event_stack.append(buy)
+    choice = self.resolve_to_choice(CardChoice)
+    self.assertEqual(choice.choices, ["Food", "Nothing"])
+    choice.resolve(self.state, "Food")
+    self.resolve_until_done()
+
+    self.assertTrue(buy.is_resolved())
+    self.assertEqual(self.char.dollars, 2)
+    self.assertEqual(self.char.possessions, [food])
+    self.assertEqual(len(self.state.common), 1)
+    self.assertEqual(self.state.common[0], gun)
+
+  def testPurchaseTwoAtList(self):
+    buy = Purchase(self.char, "common", 2, keep_count=2)
+    self.char.dollars = 8
+    self.assertFalse(buy.is_resolved())
+    self.assertFalse(self.char.possessions)
+    food = items.Food()
+    gun = items.TommyGun()
+    self.state.common.extend([food, gun])
+
+    self.state.event_stack.append(buy)
+    choice = self.resolve_to_choice(CardChoice)
+    self.assertEqual(choice.choices, ["Food", "Tommy Gun", "Nothing"])
+    choice.resolve(self.state, "Tommy Gun")
+    choice = self.resolve_to_choice(CardChoice)
+    self.assertEqual(choice.choices, ["Food", "Nothing"])
+    choice.resolve(self.state, "Food")
+    self.resolve_until_done()
+
+    self.assertTrue(buy.is_resolved())
+    self.assertEqual(self.char.dollars, 0)
+    self.assertEqual(self.char.possessions, [gun, food])
+    self.assertFalse(self.state.common)
+
+  def testPurchaseTwoAtFixedDiscount(self):
+    buy = Purchase(self.char, "common", 2, keep_count=2, discount=1)
+    self.char.dollars = 8
```

Comment:
I might make this 6 dollars to validate that the check to see if the character can buy something is done using the discounted price instead of the regular price.